### PR TITLE
Add missing omitempty to the field json schema

### PR DESCRIPTION
**Pull request by sahildua2305**
Adds missing ",omitempty" to the field json schema of the pointer
fields.

Fixes: #786

### CODE REVIEW

**Comment by sahildua2305** [comment]
This is work in progress. I'm looking for more files to fix using the following regex in the entire package.
```
\*[a-z]+\s+`json:"[a-z_]+"`
```

**Comment by sahildua2305** [comment]
@shurcooL this is ready for a review.

**Comment by sahildua2305** [comment]
@shurcooL do we need any changes here?

**Comment by shurcooL** [comment]
> do we need any changes here?

I'll let you know when I get to reviewing this PR.

Thanks for your work, I'll get to it soon.

File: github/repos.go [hunk 1]
```diff
@@ -61,18 +61,18 @@ type Repository struct {
...
-	HasIssues         *bool   `json:"has_issues"`
-	HasWiki           *bool   `json:"has_wiki"`
-	HasPages          *bool   `json:"has_pages"`
+	Private           *bool   `json:"private,omitempty"`
+	HasIssues         *bool   `json:"has_issues,omitempty"`
+	HasWiki           *bool   `json:"has_wiki,omitempty"`
+	HasPages          *bool   `json:"has_pages,omitempty"`
+	HasDownloads      *bool   `json:"has_downloads,omitempty"`
 	HasProjects       *bool   `json:"has_projects,omitempty"`
-	HasDownloads      *bool   `json:"has_downloads"`
```
Comment (shurcooL) [inline github/repos.go]: What's the reason you re-arranged the order of the fields?

It's not alphabetical, and I can't quite see the pattern. If there's a good reason, that's fine. If not, it'd be better to keep their order unchanged and focus this PR on just adding `,omitempty`s.

Comment (sahildua2305) [reply github/repos.go]: No reason. That's wrong, indeed. I mistakenly changed the order of fields while resolving the merge conflict with that new field (`HasProjects`).

**Comment by shurcooL** [comment]
Merged. Please let us know if any issues come up!

**Comment by sahildua2305** [comment]
@shurcooL even I was surprised to know that. Good though :D

**Pull request merged**